regionmask/defined_regions/from_geopandas.py

```diff
@@ -0,0 +1,131 @@
+            )
+        raise ValueError("Found duplicates {}. {}".format(duplicates, to_check))
+
+
+def _check_missing(data, name):
+    if data.isnull().any():
+        raise ValueError("{name} cannot contain duplicate values".format(name))
+
+
+def _construct_abbrevs(geodataframe, names):
+    """Construct unique abbreviations based on geodataframe.names."""
+    if names is None:
+        raise ValueError(
+            "names is None, but should be a valid column name of"
+            "geodataframe, choose from {}".format(geodataframe.columns)
+        )
+    abbrevs = []
+    for item in geodataframe.T:
+        name_for_abbrev = geodataframe.loc[item][names]
+        # catch region with no name
+        if name_for_abbrev is None:
+            name_for_abbrev = "UND"  # for undefined
+        abbrev = "".join(word[0] for word in name_for_abbrev.split(" "))
+        # remove certain chars
+        for char_to_remove in ["-", " ", ")", "("]:
+            abbrev = abbrev.replace(char_to_remove, "")
+        # if find duplicates, add counter
+        counter = 2
+        if abbrev in abbrevs:
+            while (abbrev + str(counter)) in abbrevs:
+                counter += 1
+            abbrev = abbrev + str(counter)
+        abbrevs.append(abbrev)
+    return abbrevs
+
+
+def from_geopandas(
+    geodataframe, numbers=None, names=None, abbrevs=None, name="unnamed", source=None
+):
+    """
+    Create ``regionmask.Region`` from ``geopandas.geodataframe.GeoDataFrame``.
+
+    Parameters
+    ----------
+    geodataframe : geopandas.geodataframe.GeoDataFrame
+        Shapefile to be downloaded. File has extensions ``.shp`` or ``.zip``.
+    numbers : str (optional)
+        Number of the column in shapefile that gives a region its number.
+        This column shouldnt have duplicates. If None (default), takes
+        ``geodataframe.index.values``.
+    names : str
+        Name of the column in shapefile that names a region. Breaks for duplicates.
+    abbrevs : str
+        Name of the column in shapefile that five a region its abbreviation.
+        Breaks for duplicates. If ``construct``, a combination of the first letters of
+        region name is taken.
+    name : str (optional)
+        name of the ``regionmask.Region`` instance created
+    source : str (optional)
+        source of the shapefile
+
+    Returns
+    -------
+    regionmask.core.regions.Regions
+
+    """
+    # get necessary data for Regions
+
+    if numbers is not None:
+        # sort, otherwise breaks
+        geodataframe = geodataframe.sort_values(numbers)
+        # ensure integer
+        geodataframe[numbers] = geodataframe[numbers].astype("int")
+        numbers = _maybe_get_column(geodataframe, numbers)
+    else:
+        numbers = geodataframe.index.values
+    # make sure numbers is a list
+    numbers = np.array(numbers)
+
+    if abbrevs is not None:
+        if abbrevs == "construct":
```

Comment:
How should we call this? `"construct"`? `"__construct__"`? `"_from_name"`?

Reply:
like `"_from_name"` most.

another dirty workaround the worked for me, is just taking the same column for names and abbrevs.

Reply:
should I just rename the API string (what I did now) or the whole function name?

Reply:
the name of the function is fine for me